site/app/templates/admin/admin_gradeable/AdminGradeablePeers.twig

```diff
@@ -141,36 +169,157 @@
+        grader_students_div.append("Students for New Grader:<br />")
+        grader_students_div.append('<input class="add_user_ids" type="text" name="add_user_id_0" /><br>');
+        $('[name="add_user_id_0"]', form).autocomplete({
+            source: student_autofill
+        });
+        $('[name="add_user_id_0"]').autocomplete( "option", "appendTo", form );
+        grader_students_div.append('<p id="add_student_button" style="cursor: pointer;" onclick="addStudentTextbox(\'#admin-gradeable-peers-form2\',\'#new-grader-students\');" aria-label="Add More Users"><i class="fas fa-plus-square"></i>Add More Users<br></p>');
+    }
+
+    function checkNewGraderValid(user_input) {
+        // check if the new grader is a valid student by autofill
+        let student_autofill = JSON.parse($('#student-autofill').val());
+        let valid_user_ids = student_autofill.map(tmp => tmp['value']);
+        let valid_entry = valid_user_ids.includes(user_input.val());
+        // check that the new grader is not already a grader in the peer matrix
+        let existing_graders = [];
+        $(".peer-table-grader").each(function(i, selected) {
+            existing_graders[i] = $(selected).text();
+        });
+        let new_grader = !existing_graders.includes(user_input.val());
+        // if both above conditions are met, allow user to submit
+        return (valid_entry && new_grader);
+    }
+
+    /**
+    * Disables / enables the submit button based on the given boolean parameters. If the button is disabled,
+    * sets an error message for the user to see.
+    *
+    * @param {boolean} valid_grader true if the new grader specified is a valid student by autofill
+    * @param {boolean[]} valid_students with each boolean in the array representing whether a new student
+    *   to be added is valid by autofill
+    * @param {boolean} new_grader true if the submit button is for a new grader popup, false if it's for
+    *   editing an existing grader popup
+    * @param {boolean} grader_removed true if existing grader was removed
+    * @param {String} button_id the ID of the submit button to be enabled / disabled
+    * @param (String) error_div_id the ID of the error message to be displayed if submit is disabled
+    */
+    function updateSubmitButton(valid_grader, valid_students, new_grader, grader_removed, button_id, error_div_id) {
+        if (grader_removed || (valid_grader && !valid_students.includes(false) && (!new_grader || valid_students.includes(true)))) {
+            $(error_div_id).hide();
+            $(button_id).prop('disabled', false);
+        } else {
+            $(error_div_id).empty();
+            if (!valid_grader) {
+                $(error_div_id).text("Please enter a valid new grader that is not already in the matrix.");
+            } else if (valid_students.includes(false)) {
+                $(error_div_id).text("Please make sure all new students are valid students in the course.");
```

Comment:
I think it would be better if the error message would suggest whats wrong with the newly added student as there are multiple cases when the added student can be 'invalid' (grader = student, student already present, student out of autofill)...